compendium/DeclarativeServices/test/TestUpdateConfiguration.cpp

```diff
@@ -29,6 +29,78 @@
 #include "TestInterfaces/Interfaces.hpp"
 
 namespace test {
+
+TEST_F(tServiceComponent, testUpdatePropsBeforeInstallResolvesService)
+{
+  cppmicroservices::BundleContext ctx = framework.GetBundleContext();
+
+  // Get a service reference to ConfigAdmin to create the component instance.
+  auto configAdminService =
+    GetInstance<cppmicroservices::service::cm::ConfigurationAdmin>();
+  ASSERT_TRUE(configAdminService)
+    << "GetService failed for ConfigurationAdmin.";
+
+  // Create configuration object and update properties BEFORE installing and
+  // starting the bundle which defines the service.
+  auto configuration = configAdminService->GetConfiguration("sample::ServiceComponentCA02");
+  configuration->UpdateIfDifferent(
+    std::unordered_map<std::string, cppmicroservices::Any>{ { "foo", true} });
+
+  // Install and start the bundle which has the server
+  auto testBundle = ::test::InstallAndStartBundle(ctx, "TestBundleDSCA02");
```

Comment:
Doesn't this bundle need to have a "cm" section in it to accurately recreate the use case? The failure case was creating and updating a configuration and then installing and starting a bundle containing a DS service and a "cm" section.

Reply:
I think there are two use cases. In one use case the configuration object is created programmatically then the bundle is started. In the second use case the configuration object is in the manifest.json file and is created by ConfigurationAdmin automatically when the bundle is started. In both cases the service should be resolved because the configuration object is available. That's why there are two new tests. The first one creates the config object programmatically. The second creates it in the manifest.json file.

Reply:
yes, i agree, however the reported reproduction steps has the configuration being created and updated before the bundle containing the "cm" section is installed and started. The test case here does those operations in reverse and I think we should have a test case that reproduces the same reproduction steps.

Reply:
cleared this up offline. there are two separate reproduction cases. I retract my previous comment.

in addition there is one more use case that should be tested which is basically the merging of these two test cases: a bundle with services and "cm" data with the configuration being updated before the bundle is installed and started.

Reply:
Added another use case that programmatically creates the configuration object before the bundle defining the service is started. The bundle defining the service also contains the same configuration object. Confirmed the service is resolved and the resulting properties match those in the manifest.json file since that update happens last.